### PR TITLE
Fixes #14271 - Show message on empty search results

### PR DESCRIPTION
**Pull request by johnpmitsch**
On an empty search result, a message that no results are returned
will be shown instead of a message that says "No _____ exists"

### CODE REVIEW

**Comment by johnpmitsch** [comment]
Requires https://github.com/Katello/bastion/pull/116

**Comment by johnpmitsch** [comment]
Still have some work to do on this, but wanted some feedback on the design

**Comment by johnpmitsch** [comment]
@ehelms @waldenraines please give this a quick glance if you can

Also - Errata content pages seem to work off of the [same controller](https://github.com/Katello/katello/blob/master/engines/bastion_katello/app/assets/javascripts/bastion_katello/environments/details/environment-content.controller.js). How can I specify a resource name that changes for each page so I can display "No \<content name\>  results"?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-add-subscriptions.controller.js [hunk 1]
```diff
@@ -26,7 +26,12 @@ angular.module('Bastion.activation-keys').controller('ActivationKeyAddSubscripti
             'organization_id': CurrentOrganization,
             'search': $location.search().search || "",
             'sort_order': 'ASC',
-            'available_for': 'activation_key'
+            'available_for': 'activation_key',
+            'resourceName': 'Subscriptions',
+            'emptyRowsMessage': "You currently don't have any Subscriptions to subscribe to," +
+                                " you can add Subscriptions by selecting 'Products' under" +
+                                " 'Content' in the main menu"
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-add-subscriptions.controller.js]: Why move these out of the html and into this JS file?  To me these messages make more sense in the html.

Also, by doing it this way these messages are no longer translated.

Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-add-subscriptions.controller.js]: See https://github.com/Katello/bastion/pull/116#discussion_r70081611 for my suggestion on how we should do this.

**Comment by johnpmitsch** [comment]
@ehelms this will need a release of bastion to function now that https://github.com/Katello/bastion/pull/116 is merged

**Comment by johnpmitsch** [comment]
@ehelms @waldenraines Can you please take a look if this is ready to merge?

**Comment by waldenraines** [comment]
Looks good to me, ACK pending Bastion release.

**Comment by ehelms** [comment]
Should be good to go here as Bastion has been released to gem and a PR to foreman-packaging open.

**Pull request merged**